R/render.r

```diff
@@ -409,6 +417,59 @@ data_deps <- function(pkg, depth) {
 
   check_bootswatch_theme(bootswatch_theme, bs_version, pkg)
 
+  light_deps <- create_theme(pkg, bootswatch_theme, bs_version, mode = "light")
+  dark_deps <- create_theme(pkg, bootswatch_theme, bs_version, mode = "dark")
+  # only keep what's different
+  dark_deps <- dark_deps[purrr::map_chr(dark_deps, "name") == "bootstrap"]
+  dark_deps[[1]]$script <- NULL
+
+  # Dependencies files end up at the website root in a deps folder
+  deps <- c(
+    lapply(
+      light_deps,
+      htmltools::copyDependencyToDir,
+      file.path(pkg$dst_path, "deps")
+    ),
+    lapply(
+      dark_deps,
+      htmltools::copyDependencyToDir,
+      file.path(pkg$dst_path, "deps-dark")
+    )
+  )
+
+  # Function needed for indicating where that deps folder is compared to here
+  transform_path <- function(x) {
+
+    # At the time this function is called
+    # html::renderDependencies() has already encoded x
+    # with the default htmltools::urlEncodePath()
+    x <- sub(htmltools::urlEncodePath(pkg$dst_path), "", x)
+
+    if (depth == 0) {
+      return(sub("/", "", x))
+    }
+
+    paste0(
+      paste0(rep("..", depth), collapse = "/"), # as many levels up as depth
+      x
+    )
+
+  }
+  # Tags ready to be added in heads
+  htmltools::renderDependencies(
+    deps,
+    srcType = "file",
+    hrefFilter = transform_path
+  )
+}
+
+create_theme <- function(pkg, bootswatch_theme, bs_version, mode) {
+  meta <- if (mode == "light") {
```

Comment:
so the logic for customization is

* you have customization variables for light mode, that automatically are customization variables for dark mode unless you override them in the "dark-bslib" field. 
* these customization variables are combined with pkgdown defaults. Then the combined list is fed to bslib.

This way you can change fonts only once. But what's a pity is that if you use your own fancy colors for light mode, and are happy with pkgdown defaults for dark mode, you need to _unset_ all the variables for dark mode e.g.

```yaml
template:
  bslib:
    primary: green
  dark-bslib:
    primary: null
```

Reply:
of course this needs to be documented once validated.

Reply:
How about we have `bslib`, `bslib-light` and `bslib-dark`?

Reply:
that's an excellent idea. 

Note to self: make sure there's no partial matching problem.